src/mqt/yaqs/core/data_structures/hamiltonian_schedule.py

```
@@ -0,0 +1,147 @@
+        """Resolve one substep, reusing a small least-recently-used cache.
+
+        Returns:
+            The static MPO for the substep's parameter tuple.
+        """
+        key = self._cache_key(substep.parameters)
+        if key is not None and key in self._cache:
+            mpo = self._cache.pop(key)
+            self._cache[key] = mpo
+            return mpo
+
+        mpo = Hamiltonian._resolve_factories(  # ruff: ignore[private-member-access]
```

Ruff (0.16.1)

[warning] Invalid rule code in suppression: private-member-access

Enable `lint.preview` to use rule names

(RUF102)